client/components/comments/index.js

```diff
@@ -63,10 +63,10 @@ class Comments extends Component {
       >
         <Fragment>
           {
-            showPrevious && previous.map((comment, index) => <Comment index={index} key={index} { ...comment } user={this.props.user} />)
+            (showPrevious || !active.length) && previous.map((comment, index) => <Comment index={index} key={index} { ...comment } user={this.props.user} />)
           }
           {
-            !!previous.length && (
+            !!active.length && !!previous.length && (
               <p><Button className="link" onClick={this.toggleShowPrevious}>{ showPrevious ? 'Hide' : 'Show' }</Button> {previous.length} previous comments</p>
```

Comment:
Out of scope I know, but is there a reason that _only_ the hide/show part is clickable? That seems super weird to me, and contrary to all received wisdom.